src/algebraic_geometry/locally_ringed_space/has_colimits.lean

```diff
@@ -0,0 +1,113 @@
+/-
+Copyright (c) 2021 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import algebraic_geometry.locally_ringed_space
+import algebra.category.CommRing.constructions
+import algebraic_geometry.open_immersion
+import category_theory.limits.constructions.limits_of_products_and_equalizers
+
+/-!
+# Colimits of LocallyRingedSpace
+
+We construct the explict coproducts and coequalizers in the category of `LocallyRingedSpace`.
+It then follows that `LocallyRingedSpace` has all colimits, and
+`forget_to_SheafedSpace` preserves them.
+-/
+
+namespace algebraic_geometry
+
+universes v u
+
+open category_theory category_theory.limits opposite
+
+namespace SheafedSpace
+
+variables {C : Type u} [category.{v} C] [has_limits C]
+variables {ι : Type v} (F : discrete ι ⥤ SheafedSpace C)
+
+lemma sigma_ι_jointly_surjective (x : colimit F) :
+  ∃ (i : ι) (y : F.obj i), (colimit.ι F i).base y = x :=
```

Comment:
Can you use the lemma `category_theory.limits.concrete.is_colimit_exists_rep` to prove this, using the fact that the forgetful functor to `Top` preserves colimits?